lib/dbPool.js

```diff
@@ -193,38 +47,40 @@ class DBPool {
    * @param {number} index - An identifier to id the connection for debug purposes.
    */
   createConnection(index) {
-    let me = this,
-      connections = me.connections;
+    const me = this;
+    const { connections } = me;
 
     // Indices can only be > 0 if integer-based
     if (typeof index === 'undefined' || (Number.isInteger(index) && index < 0)) {
+      // eslint-disable-next-line no-param-reassign
       index = connections.length;
     }
 
     me.log(`Creating Connection ${index}...`);
     me.connections.push(new DBPoolConnection(index, {
       database: me.database,
       options: {
-        debug: me.debug
-      }
+        debug: me.debug,
+      },
     }));
     me.log(`Connection ${index} created`);
   }
 
   /**
    * Frees all connections in the pool (Sets "Available" back to true for all)
    * closes any statements and gets a new statement.
-   * @returns {boolean} - true if all were detached succesfully
+   * @returns {boolean} - true if all were detached successfully
    */
   async detachAll() {
-    let me = this,
-      connections = me.connections,
-      connection = {},
-      reason;
+    const me = this;
+    const { connections } = me;
+    // let connection = {};
+    let reason;
 
-    for (connection of connections) {
+    for (let i = 0; i < connections.length; i += 1) {
```

Comment:
Why not just `this.connections.length` instead of 
```
const me = this;
const { connections } = me;
i < connections
```
?

Reply:
updated to just use this